nautobot_circuit_maintenance/jobs/site_search.py

```diff
@@ -0,0 +1,161 @@
+    Args:
+        maintenance_queryset (Queryset): Queryset of all Circuit Maintenance objects
+
+    Returns:
+        dict: Dictionary of a set of maintenance records
+    """
+    return_dictionary = collections.defaultdict(set)
+    for record in maintenance_queryset:
+        # Get the site overall
+        # Start with looping over the circuits that are defined
+        for circuit in record.circuits:
+            # Check both termination A and Z for values of None, add that record to the set for each site
+            for term in [circuit.termination_a, circuit.termination_z]:
+                if term is not None:
+                    return_dictionary[term.site.name].add(record)
+
+    return dict(return_dictionary)
+
+
+class FindSitesWithCircuitImpact(Job):
+    """Nautobot Job definition for finding sites without redundant circuit for impactful maintenance.
+
+    Args:
+        Job (Nautobot Job): Nautobot Job import.
+    """
+
+    class Meta:
+        """Meta definition for the Job."""
+
+        name = "Find Sites With Circuit Impact"
+        description = "Search for sites with multiple circuits, 1 or more circuit impacts."
+        read_only = True
+
+    def run(self, data=None, commit=None):
+        """Executes the Job.
+
+        Args:
+            data (_type_, optional): _description_. Defaults to None.
+            commit (_type_, optional): _description_. Defaults to None.
+        """
+        # Query for all of the circuits maintenances that are on going in the future
+        today = date.today()
+        circuit_maintenances = CircuitMaintenance.objects.filter(start_time__gte=today).order_by("start_time")
+
+        # Build a circuit mapper
+        circuit_maintenance_mapper = build_sites_to_maintenance_mapper(circuit_maintenances)
+
+        # Loop over each of the circuit maintenance records
+        for ckt_maint in circuit_maintenances:
+            # Get the list of sites
+            for circuit in ckt_maint.circuits:
+                sites = get_sites_from_circuit(circuit)
+
+            # Check to see if there are any circuit maintenances that are duplicated time
+            for site in sites:
+                for other_ckt_maint in circuit_maintenance_mapper[site.name]:
+                    # Report failures for any time where a circuit will take an outage
+                    if ckt_maint == other_ckt_maint:
+                        self.log_info("Found each other!")
+                        continue
+                    if check_for_overlap(ckt_maint, other_ckt_maint):
+                        self.log_warning(
+                            obj=site,
+                            message=f"There is an overlapping maintenance for site: {site.name}. Other maintenances: {other_ckt_maint}",
+                        )
+                    else:
+                        # Log success for each time there is a known circuit still available at the site at the same time
+                        self.log_success(
```

Comment:
I would leave this as `debug`, I believe this should be the "default"

Reply:
Why keep it as debug versus success? I would like to see green checks to help see that there was a check and no overlap was found.

Reply:
My understanding of "success" is not about the network status health, but the health of the Job execution. So, from the Job point of view, this is just an info step that it has been processed and evaluated. Even when there are overlaps, the overlal Job should "success" and show the report with the issues.

Reply:
Warning is good enough for overlappings

Reply:
Went with `log_info` instead of `log_success`. And will be adding a log_success at the bottom